packages/types/src/authentication.ts

```diff
@@ -1,7 +1,36 @@
 import * as z from 'zod';
+import { UserSummary } from '.';
 
-export const SignUpResponseBody = z.object({
-    userID: z.string().uuid(),
-    userNickname: z.string(),
+export const AuthenticationModeValues = z.enum(['web', 'api']);
+export type AuthenticationModeValues = z.infer<typeof AuthenticationModeValues>;
+
+export const SignUpSuccessfullResponseBody = z.object({
+    token: z.string().optional(),
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** Maybe will we find interesting, when dealing with front-end code, to have two response types, one for web authentication, with no token, and another one for api authentication, with a mandatory token?

Reply:
I was also wondering on this point
But pragmatically there's no need to make the distinction if it's not for global comprehension that is 100% a good motivation
I mean that web auth doesnot implies any more code in the frontend, the server will manage it for us. In this way the client can consider that if the token is set then it needs to be stored as it involves an api auth ?
I'm not sure about this way of thinking, but it's still was the one I adopted :laughing:

Reply:
I think that in the client we will make a clear distinction between web and api authentications, in term of processes to execute depending on the platform.
If we are on the web, we ask for web authentication and expect no token.
If we are on a native platform, we ask for api authentication and expect a token, that we store.
Having strong types would make the contract between frontend and backend even stronger.

Reply:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought:** Should we have two distinct routes that could hit the same method by precomputing the authenticationMode in this way we could strongly type specific route responseBody or is it overkill ?
Indeed I wasn't looking it this way, by platform. You're right

Reply:
> thought_balloon thought: Should we have two distinct routes that could hit the same method by precomputing the authenticationMode in this way we could strongly type specific route responseBody or is it overkill ?

I mean It's overkill, typescript and the client signUp service itself will do the job.

Reply:
I think the implementation you made of the server route is great!

Reply:
Updated ! in last commits